### ISSUE TITLE
Serve all library files through the moodle file serving api

### ISSUE THREAD

**Issue by brendanheywood** [h5p/moodle-mod_hvp#389]
This is more along the lines of https://github.com/h5p/moodle-mod_hvp/issues/338 and https://github.com/h5p/moodle-mod_hvp/issues/339, there is a large number of requests for files which are not cached properly:

```
    182 "GET /mod/hvp/library/styles/h5p.css?ver=2020080401
    181 "GET /mod/hvp/library/styles/h5p-confirmation-dialog.css?ver=2020080401
    181 "GET /mod/hvp/library/js/jquery.js?ver=2020080401
    179 "GET /mod/hvp/xapi-collector.js
    178 "GET /mod/hvp/library/styles/h5p-core-button.css?ver=2020080401
    178 "GET /mod/hvp/library/js/h5p.js?ver=2020080401
    176 "GET /mod/hvp/library/js/h5p-x-api-event.js?ver=2020080401
    176 "GET /mod/hvp/library/js/h5p-event-dispatcher.js?ver=2020080401
    174 "GET /mod/hvp/library/js/request-queue.js?ver=2020080401
    174 "GET /mod/hvp/library/js/h5p-x-api.js?ver=2020080401
    173 "GET /mod/hvp/library/js/h5p-content-type.js?ver=2020080401
    172 "GET /mod/hvp/library/js/h5p-confirmation-dialog.js?ver=2020080401
    172 "GET /mod/hvp/library/js/h5p-action-bar.js?ver=2020080401
    157 "GET /mod/hvp/library/fonts/h5p-core-24.ttf?mz1lkp
    152 "GET /mod/hvp/library/js/h5p-resizer.js
```

These should all get served via the moodle serving api (but not stored in the File API), so that moodle can add the cache control headers so these get cached by upstream cdn's

**Comment by barry-matthias** [comment]
I agree with this bug the h5p module stores approx 3370 js files in the lms file system which are not cached as they should be. This significantly slows down the loading times for users as on average 1000 files need to be served to meet the url request received by the lms for certain content such as interactive video. The loading times then get compounded if the number of users requesting the same content and hence the same files increases. The issue is only mitigated when the files are finally loaded and cached locally to the users browser.

**Comment by lucaboesch** [comment]
Dear @brendanheywood is this report still up to date?
Are those file still served that way?

At least, I do see https://…/lib/javascript.php/1787597035/mod/hvp/xapi-collector.js on my site, now.
And on repeated load, it is served with HTTP 304 (Not Modified) redirection response.

Thanks for a quick feedback.

Best,
Luca

**Comment by brendanheywood** [comment]
We definitely fixed this in our fork years ago but I'm not sure if it was accepted stream. I can't tell from our logs if its still an issue because we are running our fork.

From a quick scan of https://github.com/h5p/moodle-mod_hvp/blob/stable/lib.php#L243 I would say this is not done fixed yet. The best way to tell would be access log stats looking for anything which is static and making sure that it is served with proper headers to its cached at the cdn / reverse proxy layer.

/lib/javascript.php/1787597035/mod/hvp/xapi-collector.js - yes this looks good but its the js in the hvp plugin, which is different to the js inside library files. Both should be served through one of the moodle endpoints

**Comment by otacke** [comment]
@lucaboesch Is that piece of information something that you can work with?

**Comment by otacke** [comment]
@lucaboesch ping

**Comment by otacke** [comment]
@lucaboesch pong
